dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -105,30 +105,41 @@ def refute_estimate(self):
         self.logger.info("Refutation over {} simulated datasets".format(self._num_simulations) )
         self.logger.info("The transformation passed: {}".format(self._transformations) )
 
-        data_chunks = self.preprocess_data_by_treatment()
+        groups = self.preprocess_data_by_treatment()
         estimates = []
-        for chunk in data_chunks:
-
-            X_input = chunk[self._chosen_variables]
-            new_outcome = chunk['y']
-            X = self._data[self._chosen_variables]
+        for key_train, _ in groups:
+            X_train = groups.get_group(key_train)[self._chosen_variables].values
+            new_outcome_train = groups.get_group(key_train)['y'].values
+            validation_df = [] 
+            for key_validation, _ in groups:
+                if key_validation != key_train:
+                    validation_df.append(groups.get_group(key_validation))
+
+            validation_df = pd.concat(validation_df)
+            X_validation = validation_df[self._chosen_variables].values
+            new_outcome_validation = validation_df['y'].values
 
             for action, func_args in self._transformations:
 
                 if callable(action):
-                    estimator = action(X_input, new_outcome, **func_args)
-                    new_outcome = estimator(X)
+                    estimator = action(X_train, new_outcome_train, **func_args)
+                    new_outcome_train = estimator(X_train)
+                    new_outcome_validation = estimator(X_validation)
                 elif action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
-                    estimator = self._estimate_dummy_outcome(func_args, action, new_outcome, X_input)
-                    new_outcome = estimator(X)
+                    estimator = self._estimate_dummy_outcome(func_args, action, new_outcome_train, X_train)
+                    new_outcome_train = estimator(X_train)
+                    new_outcome_validation = estimator(X_validation)
                 elif action == 'noise':
-                    new_outcome = self._noise(new_outcome, func_args)
+                    new_outcome_train = self._noise(new_outcome_train, func_args)
+                    new_outcome_validation = self._noise(new_outcome_validation, func_args)

```

Comment:
shouldn't the _noise method defined on line 293 take a parameter that is the fraction_Std_dev to denote how much noise to add?

Reply:
Also, shouldn't this be passed as **func_args given the def in 293?

Reply:
Some functions, such as noise or permute take in a single value, in that case, is it prudent to pass a dictionary as **func_args, rather than just func_args?